code/modules/mob/living/carbon/carbon_defense.dm

```diff
@@ -284,11 +284,11 @@
 		Knockdown(stun_duration)
 
 /mob/living/carbon/proc/help_shake_act(mob/living/carbon/helper)
-	if(on_fire)
-		to_chat(helper, span_warning("You can't put [p_them()] out with just your bare hands!"))
+	if(SEND_SIGNAL(src, COMSIG_CARBON_PRE_MISC_HELP, helper) & COMPONENT_BLOCK_MISC_HELP)
 		return
 
-	if(SEND_SIGNAL(src, COMSIG_CARBON_PRE_MISC_HELP, helper) & COMPONENT_BLOCK_MISC_HELP)
+	if(on_fire)
+		to_chat(helper, span_warning("You can't put [p_them()] out with just your bare hands!"))
```

Comment:
this absolutely is a player facing change - you can now do CPR on someone who's on fire lol

just add another signal. 
```suggestion
/mob/living/carbon/proc/help_shake_act(mob/living/carbon/helper)

	if(SEND_SIGNAL(src, COMSIG_CARBON_ATTEMPT_MISC_HELP, helper) & COMPONENT_BLOCK_MISC_HELP)
		return	
	if(on_fire)
		to_chat(helper, span_warning("You can't put [p_them()] out with just your bare hands!"))
		return

	if(SEND_SIGNAL(src, COMSIG_CARBON_PRE_MISC_HELP, helper) & COMPONENT_BLOCK_MISC_HELP)
		return	

```